src/crud.js

```diff
@@ -4,24 +4,66 @@ import _ from 'lodash';
 import { parseInclude, parseWhere } from './utils';
 import { notFound } from 'boom';
 
-let prefix;
-let defaultConfig;
-
-export default (server, model, options) => {
-  prefix = options.prefix;
-  defaultConfig = options.defaultConfig;
-
-  list(server, model);
-  get(server, model);
-  scope(server, model);
-  create(server, model);
-  destroy(server, model);
-  destroyAll(server, model);
-  destroyScope(server, model);
-  update(server, model);
+const createAll = ({server, model, prefix, config}) => {
+  Object.keys(methods).forEach((method) => {
+    methods[method]({server, model, prefix, config});
+  });
+};
+
+/*
+The `models` option, becomes `permissions`, and can look like:
+
+```
+models: ['cat', 'dog']
+```
+
+or
+
+```
+models: {
+  cat: ['list', 'get']
+  , dog: true // all
+}
+```
+
+*/
+
+export default (server, model, {prefix, defaultConfig: config, models: permissions}) => {
+  const modelName = model._singular;
+
+  if (!permissions) {
+    createAll({server, model, prefix, config});
+  }
+  else if (Array.isArray(permissions) && permissions.includes(modelName)) {
+    createAll({server, model, prefix, config});
+  }
+  else if (_.isPlainObject(permissions)) {
+    const permittedModels = Object.keys(permissions);
+
+    if (permissions[modelName] === true) {
+      createAll({server, model, prefix, config});
+    }
+    else if (permittedModels.includes(modelName)) {
+      if (Array.isArray(permissions[modelName])) {
+        permissions[modelName].forEach((method) => {
+          methods[method]({server, model, prefix, config});
+        });
+      }
+      else if (_.isPlainObject(permissions[modelName])) {
+        permissions[modelName].methods.forEach((method) => {
+          methods[method]({
+            server,
+            model,
+            prefix,
+            config: permissions[modelName].config || config,
```

Comment:
Do you think we should completely override the default config instead of `_.defaults`ing it?